website_sale_checkout_store/controllers/main.py

```diff
@@ -57,3 +44,20 @@ def payment_get_status(self, sale_order_id, **post):
         else:
             return super(website_sale, self).payment_get_status(sale_order_id, **post)
 
+    def checkout_form_validate(self, data):
+        cr, uid, context = request.cr, request.uid, request.context
+        order = request.website.sale_get_order(force_create=1, context=context)
+        if order.buy_way:
+            if 'nobill_noship' in order.buy_way:
+                website_sale.mandatory_billing_fields = ["name", "phone", "email"]
```

Comment:
mandatory_billing_fields is also used in checkout_parse. You have to update it there too